packages/taquito/src/operations/batch-operation.ts

```diff
@@ -58,7 +59,8 @@ export class BatchOperation
         .filter((result) => BATCH_KINDS.indexOf(result.kind) !== -1)
         .map((result) => {
           if (hasMetadataWithResult(result)) {
-            return result.metadata.operation_result.status;
+            const opResult = result.metadata.operation_result as BatchOperationResult;
```

Comment:
casting is necessary still since opResult is type unknown. TS will also throw an error if you only type it as such. Same comment as the other one

Reply:
the operation_result type seems lost in `hasMetadataWithResult` function, will we consider defined it better in the function so we can avoid casting

Reply:
TS unfortunately doesn't derive the subset of types from `hasMetadataWithResult`. It will always return unknown regardless, which is the reason why we needed to cast it

Reply:
I understand :)